### PR TITLE
Add DiagnosticDescriptorHelper to set HelpURI for generator / analyzer diagnostics

### LINKED ISSUE #85181: F1 on our source generator errors goes nowhere because we don't set a HelpUri
I got this
image

clicked on the SYSLIB1052 code, which is hyperlinked, and got an Oops page:

https://learn.microsoft.com/en-us/visualstudio/ide/not-in-toc/default?f1url=%3FappId%3DDev16IDEF1%26l%3DEN-US%26k%3Dk(keywords_removed)%26rd%3Dtrue&view=vs-2022

Not sure if it's just this code.

### PR DESCRIPTION
**Pull request by ericstj**
Fixes #85181 

Create a helper method that formats and sets the helpLink for DiagnosticDescriptors

See https://github.com/search?q=repo%3Adotnet%2Fruntime+%28%22new+DiagnosticDescriptor%28%22+OR+%22new+DiagnosticDescriptor+%28%22%29&type=code

Absent are those from https://github.com/dotnet/runtime/blob/79b970739df6ff8d75ff7f4f262a8b2785c20940/src/tools/illink/src/ILLink.RoslynAnalyzer/DiagnosticDescriptors.cs#L14

I wasn't sure if those are published at the same URL.  Maybe those are all here: https://learn.microsoft.com/en-us/dotnet/core/deploying/trimming/trim-warnings/il2001 ?  If so - perhaps we can just modify that existing helper class.   Thoughts @mateoatr @vitek-karas?

### CODE REVIEW

**Review by eiriktsarpalis — APPROVED** [review]

**Comment by ericstj** [comment]
> If so - perhaps we can just modify that existing helper class. Thoughts @mateoatr @vitek-karas?

I took another look this morning and found that we could fix this.  Please review @dotnet/illink

File: src/tools/illink/src/ILLink.Shared/DiagnosticId.cs [hunk 1]
```diff
@@ -235,5 +235,8 @@ public static class DiagnosticIdExtensions
 				>= 3050 and <= 6000 => DiagnosticCategory.AOT,
 				_ => throw new ArgumentException ($"The provided diagnostic id '{diagnosticId}' does not fall into the range of supported warning codes 2001 to 6000 (inclusive).")
 			};
+
+		public static string GetHelpUri(this DiagnosticId diagnosticId) =>
+			$"https://learn.microsoft.com/en-us/dotnet/core/deploying/trimming/trim-warnings/il{(int) diagnosticId}";
```
[the remark this answers is not in the document]
Comment (ericstj) [reply src/tools/illink/src/ILLink.Shared/DiagnosticId.cs:240]: Can that be determined by the error code range?

[the remark this answers is not in the document]
Comment (ericstj) [reply src/tools/illink/src/ILLink.Shared/DiagnosticId.cs:240]: Based on what I see in the URLs anything 3050 or higher is `native-aot`.

**Review by sbomer — APPROVED** [review]
Thank you, LGTM!

**Comment by ericstj** [comment]
Linker test failures are https://github.com/dotnet/runtime/issues/90431

**Pull request merged**